gcp-vertex-ai-mojo-scorer/src/main/java/ai/h2o/mojos/deploy/gcp/vertex/ai/controller/ModelsApiController.java

```diff
@@ -74,8 +86,45 @@ public static ScoreRequest getRestScoreRequest(
     request.setIdField(gcpRequest.getParameters().getIdField());
     request.setFields(gcpRequest.getParameters().getFields());
     
+    // Check if a pre-processing script was provided, if so, use it first
+    Map<String, String> env = System.getenv();
+    String preProccessingScript = env.getOrDefault("PREPROCESSING_SCRIPT_PATH", "");
+    
+    if (!preProccessingScript.isEmpty()) {
+      // Write request body to JSON file to be consumed by Python script
+      ObjectMapper mapper = new ObjectMapper();
+      
+      String fileName = UUID.randomUUID().toString();
+      
+      try {
+        mapper.writeValue(new File("/tmp/" + fileName), gcpRequest);
+        
+        // Run Python pre processing script and reload request body with updated data
+        ProcessBuilder processBuilder;
+        processBuilder = new ProcessBuilder("python", "/tmp/preprocessing_script.py", fileName);
+        processBuilder.redirectErrorStream(true);
+        log.info("Waiting for script to run");
+        Process process = processBuilder.start();
+        process.waitFor();
+        
+        // Read preprocessed data
+        JsonReader reader = new JsonReader(new FileReader("/tmp/" + fileName));
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        gcpRequest = gson.fromJson(reader,
+            ai.h2o.mojos.deploy.common.rest.vertex.ai.model.ScoreRequest.class);
+        
+        Files.deleteIfExists(Paths.get("/tmp" + fileName));
```

Comment:
so I'd probably advise to to a try-with-resources block here: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

probably makes sense to always clean up and delete the file.... for example, what happens if there is an exception during the process. then the file wouldn't always be deleted and you could end up with extra cruft lying around the system taking up space

Reply:
also, there is a possibility of accidentally leaving open readers/writers which can cause problems in garbage cleanup/memory